### ISSUE TITLE
500 Rate limited errors for images with large number of tags

### ISSUE THREAD

**Issue by philwinder** [weaveworks/flux#410]
The sock shop now has hundreds of tags, due to pushing a container for every commit.

When doing a `list-images` on one of the components returns an error. I thought that the rate limiting fixed this? 

To reproduce:
On master `f1c552c` or greater.

Create a standalone setup in minikube.

```
$ kubectl create ns sock-shop ; kubectl create -f https://raw.githubusercontent.com/microservices-demo/microservices-demo/master/deploy/kubernetes/manifests/catalogue-dep.yaml; kubectl create -f https://raw.githubusercontent.com/microservices-demo/microservices-demo/master/deploy/kubernetes/manifests/catalogue-svc.yaml

$ fluxctl list-images --service sock-shop/catalogue
Problem communicating with the service:  500 Internal Server Error (getting images for services: fetching image metadata for weaveworksdemos/catalogue: Get https://index.docker.io/v2/weaveworksdemos/catalogue/manifests/30d38e8f1a56bb2cf9094a7032b7b4e7b027bcac: net/http: request canceled)
Run 'fluxctl list-images --help' for usage.
```

The fluxsvc logs show lots of:
```
ts=2017-01-30T15:54:49Z caller=registry.go:99 instanceID=<default-instance-id> component=registry registry-metadata-err="Get https://index.docker.io/v2/weaveworksdemos/catalogue/manifests/fc1b123a687322604d0e71c0810a4f3af3b75042: net/http: request canceled while waiting for connection"
```

**Comment by paulbellamy** [comment]
Only if you've deployed and configured a memcached for fluxsvc to use.

**Comment by paulbellamy** [comment]
See this for how to use memcached for fluxsvc:
https://github.com/weaveworks/service-conf/blob/master/k8s/local/fluxy/fluxsvc-dep.yaml#L22

**Comment by philwinder** [comment]
@paulbellamy It is there? https://github.com/weaveworks/flux/blob/master/deploy/standalone/flux-deployment.yaml#L21

**Comment by philwinder** [comment]
New information. This also happens in prod. However, if you repeatedly keep requesting the images, it will eventually work.

So I think the issue is that it’s just too much to request to cache in one go. If you keep trying, after a few tries it has cached all the tags and bingo, it works.

And then 20 minutes later it rm’s the cache, so it has to do that again...

Potential fixes:
- Rate limiting on our side.

**Comment by philwinder** [comment]
I believe this is due to the default 10 timeout in the request for tags/manifests.

**Comment by philwinder** [comment]
Will be fixed when we fix #389
